### ISSUE TITLE
UI to set whether a ware might be transported from/to this port

### ISSUE THREAD

**Issue by bunnybot** [widelands/widelands#1935]
The user DaggeTeo wrote in the Seafaring topic in our forums:

Another thing I have noticed with the seafaring that I would like to get an overlook is an ability turn certain wares in a port to non transportable, either sending, receiving or even both.

I like this idea. So in basics each ware in each port would have four options: "default" (no icon), "do not receive", "do not send", "no exchange at all"

------------------------------------
Imported from Launchpad using lp2gh.

 * date created: 2012-04-21T12:01:01Z
 * owner: nasenbaer
 * the launchpad url was https://bugs.launchpad.net/bugs/986546

**Comment by bunnybot** [comment]
(by wl-zocker)
The topic Nasenbaer refered to can be found under http://wl.widelands.org/forum/topic/503/?page=7

The idea sounds useful (an example can be found in the topic), one could then group a certain industry (e.g. bread) and in that way. The transportation would than be more efficient.

Where will the additional buttons be put? I suggest them to put into the dock menu because the option only concerns wares that are delivered by ships. The problem is that the user does not see it then immediately when he clicks the port. But I do not think that that is a big problem because the option will be set probably only once.

**Comment by bunnybot** [comment]
(by daggeteo)
+1

I think that this would enhance the capabilities of using seafaring in the gameplay.

**Comment by bunnybot** [comment]
(by ad-simplypeachy)
A sensible suggestion. +1

**Comment by bunnybot** [comment]
(by sirver)
I see the usefulness in this, but I feel it is quite hard to make a sensible UI for this which also incorportates the current "move out of warehouse" and "prefered to be stored here". But a solution must be found, imho: think hard people.

**Comment by bunnybot** [comment]
(by daggeteo)
#4
I agree, incorporating this without it getting confusing and cluttery is important.

My suggestion (off the top of my head...) is to make two tabs for which you control the flow.

Tab 1 - Receiving rules
Tab 2- Sending rules

The buttons would be for respective tab:
Default: no icon
Allowed - a green cross 
Not allowed - a red cross 

I believe that this approach would allow for control of transport direction.

The downside is that this would introduce two clicks for each ware that you don't want to pass through the port at all. Solution to this is to add a green/red cross button in the main window where your enable disable transportation both receiving and sending. This last part might be too cluttery, but it's food for thought!

**Comment by bunnybot** [comment]
(by sirver)
Setting to incomplete for bug sweeping.

**Comment by bunnybot** [comment]
(by wl-zocker)
I still like this idea - sometimes, my ships behave quite dim-witted and transport wares unnecessarily long and far.
I have no really good idea for the UI though. The idea in comment #5 (having two different tabs for sending and receiving) seems too complicated. My idea is the following: In the "warehouse menu" (the first two tabs), the symbols should stay as they are (green arrow up = store preferred here; red cross = do not store here; red arrow down = release all wares). In the dock menu (the last two tabs), the new icons should be introduced. If we limit on allow/forbid sending/receiving wares, there are four cases:
- Both allowed. Do not show an icon. This behavior is the default.
- Sending not allowed. No idea for the icon
- Receiving not allowed. The icon could be a red cross because this behavior is similar to the "do not store here" behavior known from warehouses.
- Neither sending nor receiving of wares allowed. No idea for the icon.
These icons should only affect the wares transported by ships. The normal warehouse function should not be influenced by the dock settings. But the warehouse settings should affect both wares transported by ships and on roads.

**Comment by bunnybot** [comment]
(by sirver)
Setting to incomplete for bug sweeping.

**Comment by bunnybot** [comment]
(by janitor)
[Expired for widelands because there has been no activity for 60 days.]

**Comment by Noordfrees** [comment]
IMHO this is pretty pointless. The pathfinder find the shortest path for a ware so disallowing ware types to go through a certain portdock would mean an increase in transport time, especially as the new shipping algorithms greatly penalize long detours. -1

**Comment by Noordfrees** [comment]
No reply for over 1½ years, closing as wontfix
